backend.py

```diff
@@ -440,6 +441,22 @@ def performInstallation(answers, ui_package, interactive):
                                "latest dedicated ISO." %
                                (answers['linstor-version'], ', '.join(available_linstor_versions)))
 
+    # if upgrading a host with xapi, check we can upgrade it first
+    if answers['install-type'] == INSTALL_TYPE_REINSTALL and answers['xapi-version']:
+        available_xapi_versions = repository.listPackagesFromRepos(
+            main_repositories, 'xapi-core', '%{evr}')
+        if not available_xapi_versions:
+            raise RuntimeError("No xapi package found")
+        available_xapi_version = available_xapi_versions[0]
+        for v in available_xapi_versions[1:]:
+            if rpm.labelCompare(('pkg', 'arch', v), ('pkg', 'arch', available_xapi_version)) >= 0:
+                available_xapi_version = v
+        if rpm.labelCompare(('pkg', 'arch', available_xapi_version), ('pkg', 'arch', answers['xapi-version'])) < 0:
+            raise RuntimeError("Cannot upgrade the host: the xapi version on the host (%s) "
+                               "is newer than the xapi version to install (%s). "
+                               "Please make sure to use the latest ISO." %
```

Comment:
- "latest ISO" may not be sufficient if user has upgraded his XS to a newly-released XAPI
- we likely won't release new ISO for every XAPI update, so we'll likely direct users to installing from HTTP to get the latest version
- even then, the message(s) to use will need to be carefully crafted

Maybe use a list of suggestions like those?  @stormi surely has more ideas
- (don't show when already installing from official URL) "try installing package from the network for a more recent version"
- wait for a new xapi package and refrain from updating in on your XS host in the meantime

Reply:
Regarding the Latest ISO, I agree with Yann.

Regarding installing from HTTP, at the moment, we don't support installing from the `base`+`updates` repos, and only update netinstall repos at the same time that we release new ISOs. We'd like to do better in the future, but that requires CI improvements, so that the installer is always tested with the latest packages, and possibly installer improvements too (Maybe there's a way to make it work as is, but it's untested). 

As there can be various situations in which this message is displayed, let's just state what we know for sure and tell users to refer to the official upgrade documentation.

(Which means that we have a documentation update to prepare so that this error is documented in it...)

```suggestion
                               "Please refer to the XCP-ng upgrade documentation." %
```

Reply:
> Regarding installing from HTTP, at the moment, we don't support installing from the base+updates repos

That would be easily solved by generating self-sufficient repos instead of having them stacked on one another